Applications/Spire/Source/Ui/ScrollableListBox.cpp

```diff
@@ -0,0 +1,248 @@
+#include "Spire/Ui/ScrollableListBox.hpp"
+#include <QCoreApplication>
+#include <QResizeEvent>
+#include <QHBoxLayout>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/ValueModel.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/ScrollBar.hpp"
+
+using namespace Spire;
+using namespace Spire::Styles;
+
+ScrollableListBox::ScrollableListBox(ListView* list_view, QWidget* parent)
+    : ScrollBox(make_body(), parent)
+    , m_list_view(list_view) {
+  m_list_view->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  m_list_view->set_edge_navigation(ListView::EdgeNavigation::CONTAIN);
+  auto layout = [=] () -> QBoxLayout* {
```

Comment:
Only capture by value when a closure outlives the lexical scope of its lambda expression.

Reply:
The is_horizontal_layout function and m_body used in the lambda is this->is_horizontal_layout() and this->m_body, so [=] implicitly captures this. I am confused what we should use to capture this, [=] or [&]?

Here is a post of "Capture *this in lambda expression":
https://www.nextptr.com/tutorial/ta1430524603/capture-this-in-lambda-expression-timeline-of-change